Show Entity task/note tabs

brendanlaschke opened the pull request:
First proposal of showing a separate tasks panel - may need some changes (especially as the timeline part is not changed for now). 

TODO/Questions:
- [x] Once #1267 has been merged the add button can be added and then the creation of activity tests can be fixed
- [x] Which task should be shown? Currently it's handled like on the notes page with Today, Upcoming, unscheduled (only not done tasks). 
- [x] Which notes are shown? Currently all.
- [x] Should we create tests for these tab components and in which form? 


<img width="977" alt="Bildschirmfoto 2023-08-19 um 13 49 43" src="https://github.com/twentyhq/twenty/assets/48770548/8b271e07-47f3-4ad0-b048-ee55f3db8f9d">

<img width="1305" alt="Bildschirmfoto 2023-08-23 um 22 20 37" src="https://github.com/twentyhq/twenty/assets/48770548/6daf9fdd-2733-4798-bee4-86571c5e6f94">


closes #1279, closes #1280, closes #1281

charlesBochet commented:
Hi @brendanlaschke, sorry for the slow response time, it has been quite busy those days!

Thanks for taking on this one!

High level:
1) It's great!

2) The activities module is getting big and it's getting harder to understand component hiearchy. Can we re-organize it in: 
activities/tasks + activities/notes + activities/timeline + activities/right-drawer ? (not related to your PR but I think we should move existing components from activities/components and activities/hooks and activities/states into these subfolders. It's a refactoring we can do later

3) you have introduce hooks useEntityTasks, useEntityNotes. These are similar to useTasks. Can we merge them together so we avoid having duplicated code? useActivities with some options should work (maybe I've missed something)

4) you are adding code to the ui/layout folder regarding Notes and Tasks. These components should live in the acitivities module and the layout should only be re-usable layout components that don't know about activities. This way we don't create a dependency from ui/layout to activities.

5) Regarding your questions, I think you are showing the right tasks/notes. As of now, we want basically the same behavior as on Tasks page but scoped to the shown entity

6) Regarding tests, here is our FE storybook testing strategy:
- what is in @/ui should be tested in depth as these are our basic building blocks. We are quite exhaustive here (see Catalog tests in Chip for example)
- what is in module should be reasonably well tested, not necessarily independently as modules are often coupled to other modules but if you feel that there is a complex business logic that needs to be tested, it might be a good idea. Most of the module code is actually tested through pages testing (see next point)
- what is in pages should be tested from a high level perspective (make sure it renders and that basic behaviors are in place)
- we also have jest tests for lower level functions like utils

So for your PR, I would add tests on Activities page and make sure that Notes, Tasks, Timeline are properly loaded, click on Open a Task, Add a Task, Open a Note, add A  note and make sure that the right drawer is opening and that's it
For your changes on Tabs for example, I would make sure that these changes are reflected in storybook so we can catch any regressions through chromatic pixel diff

Thanks again!

brendanlaschke commented:
Tried tackling the points ...

To 3 im merged the two `useTasks` but im not sure if it is useful to merge tasks and notes. They seem very different:
-  the tasks returns three lists, the notes one
- the filter specific query is for now only used with the tasks - so no need to use it when only loading notes
But if you have a good idea im happy to see it :) 

Added the tests and moved the components to the activities module (and restructured it to have `/tasks` and `/notes`).

charlesBochet reviewed (approved):
LGTM, that's a good one, thank you @brendanlaschke!
